apps/src/blockly/addons/cdoBlockFlyout.ts

```diff
@@ -1,19 +1,27 @@
-import GoogleBlockly from 'blockly/core';
+import GoogleBlockly, {IDraggable, Options} from 'blockly/core';
+import {FlyoutItem} from 'blockly/core/flyout_base';
+import {Svg} from 'blockly/core/utils';
 
 const svgPaths = GoogleBlockly.utils.svgPaths;
+interface CdoBlockFlyoutOptions extends Options {
+  minWidth: number;
+  maxWidth: number;
+}
+
 export default class CdoBlockFlyout extends GoogleBlockly.HorizontalFlyout {
+  private svgClipPath_: SVGElement | undefined;
+
   /**
    * This is a customized flyout class that extends the HorizontalFlyout class.
    * This flyout is intended to be placed inside of a block's FieldFlyout.
    *
-   * @param {Object} workspaceOptions - The options for constructing the class.
+   * @param workspaceOptions - The options for constructing the class.
    */
-  constructor(workspaceOptions) {
+  constructor(workspaceOptions: CdoBlockFlyoutOptions) {
     super(workspaceOptions);
-    this.horizontalLayout_ = true;
+    this.horizontalLayout = true;
```

Comment:
I was wondering how this was working now if it had been renamed so I inspected the field and saw this:
<img width="689" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/43474485/0f9a592d-5457-4329-ba22-7cc102bb869b">
 It looks like it's already being set by the parent class here, fortunately: https://github.com/google/blockly/blob/a517317b77c1e58727b748a99f9ae70b2c27d26a/core/flyout_horizontal.ts#L30

I suspect we could just delete this line instead since the old version wasn't accomplishing anything and your changed version is redundant.

Reply:
oh nice, will remove that then